src/test/java/com/virtuslab/archunit/BackgroundTaskEnqueuingTestSuite.java

```diff
@@ -146,7 +146,7 @@ public void check(JavaMethod method, ConditionEvents events) {
               String callTargetName = callTarget.getName();
               JavaClass callTargetOwner = callTarget.getOwner();
               if (callTargetOwner.isAssignableTo(Task.Backgroundable.class) && callTargetName.equals("queue") ||
-                  callTargetOwner.isAssignableTo(GitBrancher.class) && callTargetName.equals("checkout") ||
+                  callTargetOwner.isAssignableTo(GitBrancher.class) && !callTargetName.equals("getInstance") ||
```

Comment:
Here's the core change that spurred all other fixes in this PR (although in fact, what I've first seen was the race condition on deleting branch in slide out, which caused a _branch not found_ notification to display)